.github/workflows/ci.yaml

```diff
@@ -0,0 +1,69 @@
+name: CI
+
+on:
+  pull_request:
+
+permissions:
+      id-token: write
+      contents: read
+
+env:
+    PYTEST_ADDOPTS: "--color=yes"
+
+jobs:
+  test:
+    name: Test
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out Repo
+      uses: actions/checkout@v4
+      with:
+       fetch-depth: 0
+
+    - name: Setup Env
+      uses: ./.github/actions/setup_env
+
+    - name: Test with pytest
+      id: tests
+      run: |
+        pytest --disable-warnings ./
+
+  formatting:
+    name: Formatting
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+    -   name: Check out Repo
+        uses: actions/checkout@v4
+        with:
+            fetch-depth: 0
+
+    -   name: Setup Env
+        uses: ./.github/actions/setup_env
+
+    -   name: Ruff Formatting - src
+        run: ruff format --check ./src/
```

Comment:
Is there a benefit to have several runs?

I think, as a general strategy, ruff should format everything, unless it is excluded. Therefore, perhaps running just

```shell
ruff format --check
```
should be enough?

Then, we should also describe how to run the formatting in the top-level readme.